Fixed tests, Added tests, Did some refactoring.

SpyrosKou opened the pull request.

SpyrosKou committed:
-Moved tests to package com/github/rosjava_actionlib
-Reduced visibility of ClientStateMachine and methods
-Deprecated non-enum methods to enforce enum usage
-Fixed some tests
-removed practically empty gradle.properties

SpyrosKou committed:
Added Apache commons lang3 dependency

Will be used for different purposes such as Exception handling.

SpyrosKou committed:
Remove getLatestFoalStatusInteger

-Removed to enforce ClientState enum.
-Can also Throws NullPointException

SpyrosKou committed:
Updated Client Foal Manager

-The ClientStateMachine can be reset.
-The GoalManager can reuse the existing ClientStateMachine.

-Added some documentation

SpyrosKou committed:
Refactor ActionClient.java

-Refactoring
-Replaced id.equals("") with StringUtils.isBlank("")
-Added method to add multiple actionListeners
-Made goalIdGenerator final

SpyrosKou committed:
Created ActionLibMessagesUtils.java

A common pattern with reflection is used often in Messages.
Implemented a set of generic methods that can work for most cases and refactor some of these patterns.

SpyrosKou committed:
Fixed some TestServerClientStateMachine.java tests, changed ClientStateMachine.java

-In TestServerClientStateMachine.java some tests where failing because an integer was compared with an ClientState. Fixed also ServerStateMachine to return ClientState enum instead of integers. (To enforce enum usage)

-Throw IllegalStateException in ServetStateMachine instead of Exception, to be more precise.

SpyrosKou committed:
Refactoring in GoalIdFenerator.java

SpyrosKou committed:
Comminng fixed tests

Added tests with feedback
Added RosExecutor.java that allows running tests directly from java IDE.

SpyrosKou committed:
Update build.gradle

SpyrosKou committed:
Merge branch 'kinetic' of https://github.com/SpyrosKou/rosjava_actionlib into kinetic

Commit:
-Replaced reflection based getters/setters with generic method

-Changed visibility of ActionLibMessagesUtils.java (this is an implementation detail)
-Made classes and methods final - These classes are designed to be used as components to be aggregated/part of other classes. Not to be extended.

Commit:
Fixed 2 tests that had an issue

fmessmer commented:
thanks for the contribution
it might take a bit for me to review and test this PR

can you please elaborate a bit on which specific problems (e.g. exception) you have been facing that motivated you to provide these fixes?

SpyrosKou commented:
You are welcome.
Thank you and the other contributors for giving others the chance to use this project.

The reasons that these changes started was some tests in TestClientStateMachine.java were failing. The errors had to do with the internal states of the ClientStateMachine.
Since I plan to use it I wanted to read it and make sure it covers my needs. 

That was the motivation that got me started reading the code. I wanted to determine how to fix the tests, if it needed any fixing. As I was reading I was making some changes and got carried away.

This resulted in the following actions ( summary):

-1. Refactoring in order to enforce the use of ClientState enum. There were some points where numbers where used.
-2. I also did notice some methods doing nothing which I removed to make it easier for the API user. 
-3. I also noticed a repeating pattern of reflection that was generalized as methods and replaced 
-4. During reading the code I reduced the visibility of some methods/classes that seemed as internal parts of rosjava_actionlib to reduce the API breadth. 
-5. Eliminated some potential NPEs in comparison and getters.
-6 In the process I added some dependencies for logging and utilities.
-7. I declared as final methods/ classes that seemed they should only be used as components and not for inheritance.
- 8. I implemented hashcode and updated equals in ActionGoal.java however this just follows the previous logic. This and more message classes that should probably be updated.
-9. Reduced ClientStateMachine.java objects creation in ClientGoalManager.java by reusing the same object.


There are perhaps other things that could be changed. E.g. finalize is used.
If it is of interest, and I do need and find the time to update these I could also make another pull request.

SpyrosKou committed:
Accumulated changes

-Added Async tests with feedback
-Added Async Client/Server with feedback test
-All tests full runnable in java (runs in windows/linux with no ROS installed)
-Logging fixes
-Code refactoring/improvements

SpyrosKou committed:
-client optimizations

SpyrosKou committed:
-Use MasterStateClient instead of "OnNewPublisher" callback. (Fixed undetected server issues in waitForActionServerToStart)

-Also get last status message in ActionClient.findStatus(GoalStatusArray) by comparing stamps
-renamed a logger (using Log4J not ROS logging) in SimpleClient test
-added some fields in ActionClient.toString()
-

SpyrosKou committed:
ignored svn folder

SpyrosKou committed:
Added test changes

SpyrosKou committed:
Add transition and event logging in ServerStateMachine exceptions

Reduced visibility of internal functionalities
Used byte instead of int for internalState, as in GoalStatus.

SpyrosKou committed:
-

-Changed ServerMachineState.getState() to byte from int
-Removed error logging try/catch clauses from ActionServer.java
-Adjusted ActionServer.java to byte usage in ServerMachineState.getState()

SpyrosKou committed:
ServerStateMachine

Instead of always throwing an exception, there is a function that handles errors completely.
Can be configured in compile time to throw exception or just log it.
Older implementation was using exception handling only to write error messages.

SpyrosKou committed:
ActionServer.java - Added function to remove ActionServerListener

SpyrosKou committed:
Enforce that callbackTargets are never null, removed null checks.

Minor syntax style changes

SpyrosKou committed:
Added support for multiple callbacks on ActionServer

-ActionServer.ServerGoal changed to static class
-Made ActionServer class and its methods final (This class is not designed for inheritance)
-Removed finalizer from ActionServer
-Minor syntax updates on ActionServer
-ActionsServer.goalTracker made concurrent and renamed

SpyrosKou committed:
renamed state to stateMachine in ActionServer.ServerGoal

SpyrosKou committed:
ActionServer minor code updates

SpyrosKou committed:
Explanation about methods accept goal and goalReceived added in ActionServerListener.java

SpyrosKou committed:
Added methods to get Action topics

One method per Goal, Cancel, Status, Result and Feedback topics has been added.
A method to get all topics that the actionServer will publish and a method to get all topics the actionServer will subscribe to.

SpyrosKou committed:
<In progress>
Adding status sink actor ref

SpyrosKou committed:
Started adding information to README.md

-Partially updated README.md

SpyrosKou committed:
Reduced ClientGoalManager visibility

ClientGoalManager should be used only by ActionClient and ActionClientFuture

SpyrosKou committed:
Removed attachListener, renamed detachListener to removeListener

SpyrosKou committed:
Added gradle wrapper

SpyrosKou committed:
Added some features and instructions in README.md

SpyrosKou committed:
Merge branch 'kinetic' of https://github.com/SpyrosKou/rosjava_actionlib into kinetic

SpyrosKou committed:
Added TODO

SpyrosKou committed:
ActionClientFuture implementation made package private.

ActionFuture interface should be used for public access

SpyrosKou committed:
Added actionServerListener to constructor, Single ActionServerListener

The ActionServerListener is a required part of the ActionServer.
Requiring it at construction time makes usage of ActionServer less error prone.
Also there is no need for multiple ActionServerListeners

SpyrosKou committed:
Removed unused, deprecated method

SpyrosKou committed:
Replaced string with constants, added final modifier in some places

SpyrosKou committed:
Update ActionClientListener.java

SpyrosKou committed:
Time is not needed for GoalId uniqueness

Current time is also contained in the Timestamp
No need to add the time in the id.

SpyrosKou committed:
Updated class description

SpyrosKou committed:
GenerateID method. Updated description and turned to void.

SpyrosKou committed:
Moved isRunning method to ClientState from ClientStateMachine

ClientState transiation list changed from LinkedList to ArrayList

SpyrosKou committed:
-Changed trace to error in important logs
-Added conditional logging clause to improve performance

SpyrosKou committed:
Closing timer on server unpublish

Some times exceptions could be thrown if the server was being unpublished and the timer tried to send a statusTick. As at this point the status publisher would be null.
-Added all statements of sendStatusTick inside a try-catch, and simplified the catch statement
-Added statusTick timer finalization

SpyrosKou committed:
More efficient and more descriptive Logging

SpyrosKou committed:
Moved to gradle 6.6.1

SpyrosKou committed:
More Efficient ActionServer

SpyrosKou committed:
Remove out directory files

SpyrosKou committed:
Minor documentation additions, removed unused Configuation file

SpyrosKou committed:
Moved to  Gradle 6.7

SpyrosKou committed:
Require not null actoin goal in ClientGoalManager constructor

SpyrosKou committed:
Reduced visibility of methods topic callback methods  (should be used only internally)

SpyrosKou committed:
Minor changes, template avadoc addition etc

SpyrosKou committed:
Changed ClientState.ERROR to ClientState.NO_GOAL

Added check before log

SpyrosKou committed:
Removed some string concatenations

SpyrosKou committed:
Logger turned into Log4j, Minor code improvements

SpyrosKou committed:
StartTasks restructured, code broke down in methods, logger turned into Log4j2

SpyrosKou committed:
Replaced Println with Logging,

SpyrosKou committed:
Test that demonstrates Futures usage and client statuses, created a future based client

SpyrosKou committed:
Marked as deprecated

SpyrosKou committed:
Made getGoalState package private, and changed accordingly the tests

SpyrosKou committed:
Update FutureBasedClient.java

SpyrosKou committed:
Removed logging of goal state

SpyrosKou committed:
Renamed to FutureBasedClientTest

SpyrosKou committed:
Minor javadoc "fixes"

SpyrosKou committed:
Documented the ActionFuture example in the FutureBasedClient

SpyrosKou committed:
added punctuation

AmandineMa committed:
Change path of eu.test.utils to eu/test/utils to compile on Eclipse

AmandineMa committed:
Change GoalID generation to have a unique ID even after restart of client

SpyrosKou committed:
Merge pull request #3 from amdia/kinetic

Change path of test folder + goal id generation

SpyrosKou committed:
Merge branch 'kinetic' of https://github.com/SpyrosKou/rosjava_actionlib into kinetic
